bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -913,37 +917,44 @@ private boolean matchStateToValue(State state, String value, String matchConditi
                     break;
             }
         } else {
-            if (DecimalType.class.isInstance(state)) {
+            if (state instanceof DecimalType || state instanceof QuantityType<?>) {
                 try {
+                    double compareDoubleValue = Double.parseDouble(unquotedValue);
+                    double stateDoubleValue;
+                    if (state instanceof DecimalType) {
+                        stateDoubleValue = ((DecimalType) state).doubleValue();
+                    } else {
+                        stateDoubleValue = ((QuantityType<?>) state).doubleValue();
```

Comment:
This unfortunately only holds true when the label says `%unit%`, so that the unit displayed for the widget is equal to the item state unit.
In case the widget label says `%.1f K` all temperature values will be converted to kelvin prior to rendering. So before taking the double value the state may have to be converted to the widget unit. [Have a look here](https://github.com/eclipse/smarthome/blob/master/bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java#L414) on how this is done for the rendering part.

Reply:
But that is exactly how we discussed it? The value colors should be in the same unit as the label. So if K is selected for the label, the coloring values should be K, too. The `state` already has the converted unit. 

`matchStateToValue? is called from `processColorDefintion`, which is called either from `getLabelColor` or `getValueColor`, on both cases the state is determined by `getState(w)` which does the conversion.

Reply:
I just looked through the code and the sitemap DSL and its a bit more complicated than that. The sitemap DSL defines a color definition to be `<item>?<condition><state>=<argument>`. So instead of the state of the item displayed by the widget it can also check on any other item's state here.
So I guess we have to convert nonetheless. When the state is already in the widget unit the conversion is a NOP.

Reply:
Yes, you are right. Conversion of _another_ item's state does not make sense here. So The code is fine and good to go.